### DISCUSSION TITLE
Dynamic RuntimeConfig?

### DISCUSSION THREAD

**Discussion by ElYaiko** [nuxt/nuxt#27118] [Questions] [answered]
I have a site that has dynamic subdomains with a nginx reverse proxy.
So in each request Nuxt renders the site with the local URL (http://localhost:3001), so I have to set the domain in the config `app.cdnURL`, the problem here is that the subdomain needs to be dynamic by the request (Like from the `origin` header), so I have no idea how to set this variable dynamically.

Already tried with modules and plugins, but none worked.
Does anyone knows how to fix this?, Thanks.

**Comment by xjccc** [comment]
![image](https://github.com/nuxt/nuxt/assets/16478776/8b185217-3bc6-4ebf-9761-9941571994ba)

On client side, you could use plugin to set runtimeConfig, its writable and reactive.

On server side, its readonly.

You could use `NUXT_` prefix, before start the app.

```
NUXT_PBULIC_DOMAIN=xxx node .nuxt/server/index.mjs
```

**Reply by xjccc** [reply, marked as answer]
Another discussion here.
https://github.com/nuxt/nuxt/discussions/27158#discussioncomment-9404294
